spec/services/charts/annotate_spec.rb

```diff
@@ -1,250 +1,791 @@
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Meh 2022')).to eq(Date.parse('01/06/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Gorff 2022')).to eq(Date.parse('01/07/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Awst 2022')).to eq(Date.parse('01/08/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Medi 2022')).to eq(Date.parse('01/09/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Hyd 2022')).to eq(Date.parse('01/10/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Tach 2022')).to eq(Date.parse('01/11/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Rhag 2022')).to eq(Date.parse('01/12/2022'))
-        I18n.locale = 'en'
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Jan 2022')).to eq(Date.parse('01/01/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Feb 2022')).to eq(Date.parse('01/02/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Mar 2022')).to eq(Date.parse('01/03/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Apr 2022')).to eq(Date.parse('01/04/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 May 2022')).to eq(Date.parse('01/05/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Jun 2022')).to eq(Date.parse('01/06/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Jul 2022')).to eq(Date.parse('01/07/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Aug 2022')).to eq(Date.parse('01/08/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Sep 2022')).to eq(Date.parse('01/09/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Oct 2022')).to eq(Date.parse('01/10/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Nov 2022')).to eq(Date.parse('01/11/2022'))
-        expect(Charts::Annotate.new(school).send(:date_for, '01 Dec 2022')).to eq(Date.parse('01/12/2022'))
+    context 'with no intervention or activity observations' do
+      it 'returns no annotations' do
+        expect(subject_multi_fuel.annotate_weekly(x_axis_categories)).to be_empty
       end
     end
 
-    describe 'annotating weekly charts' do
-
-      subject { Charts::Annotate.new(school).annotate_weekly(x_axis_categories)}
+    context 'show on charts' do
+      let!(:intervention_observation_multi_fuel) { create(:observation, :intervention, school: school, at: Date.new(2018, 6, 24), intervention_type: multi_fuel_intervention) }
+      let!(:activity_observation_multi_fuel) { create(:observation, :activity, school: school, at: Date.new(2018, 7, 8), activity: activity_multi_fuel) }
 
-      let(:x_axis_categories){ [
-        "24 Jun 2018",
-        "01 Jul 2018",
-        "08 Jul 2018",
-        "15 Jul 2018"
-      ]}
+      it 'is set true' do
```

Comment:
This is what I was trying to explain earlier - about only putting test setup in `before` statements and putting only tests in `it` statements. I would suggest something more like this:

```
context "is true" do
  before do
    multi_fuel_intervention.update(show_on_charts: true)
    activity_type_multi_fuel.update(show_on_charts: true)
   end
   it "does whatever is expected"
end
```

Reply:
:+1:

Reply:
have pushed an update for that @urbanwide